ui/src/components/query/builder/edit/datatype/DatatypeSelect.vue

```diff
@@ -54,19 +75,34 @@ watch(
   () => propertyType.value,
   () => {
     if (propertyType.value === "range" && !isObjectHasKeys(props.property, ["range"])) {
+      props.property.operator = undefined;
+      props.property.isNull = undefined;
       props.property.range = { from: {} as Assignable, to: {} as Assignable } as Range;
     } else if (propertyType.value === "startsWith" || propertyType.value === "contains") {
+      delete props.property.range;
+      props.property.isNull = undefined;
       props.property.operator = propertyType.value;
     } else if (propertyType.value === "is") {
       delete props.property.range;
+      props.property.isNull = undefined;
       props.property.operator = "=";
+    } else if (propertyType.value === "notNull") {
+      delete props.property.range;
+      props.property.operator = undefined;
```

Comment:
When I was writing this I thought of refactoring the whole delete/=undefined to a resetProperty method with an exclude? parameter resetProperty(exclude?) but never implemented it. Perhaps not as part of this pr, it would be worth implementing.

Reply:
Possibly - problem is that each case is a little different, at least in-line it's explicit what its doing.